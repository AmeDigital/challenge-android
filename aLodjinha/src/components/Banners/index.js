import React, { useState, useEffect } from 'react';
import Carousel from 'react-native-banner-carousel';
import { StyleSheet, Image, View, Dimensions, TouchableOpacity, Linking, Alert } from 'react-native';

import api from '../../services/api';
import Loading from '../Loading';

const BannerWidth = Dimensions.get('window').width;
const BannerHeight = 150;

export default function Banners() {
	const [ banners, setBanners ] = useState([]);

	useEffect(() => {
		(async function loadBanners() {
			const response = await api.get('/banner');
			setBanners(response.data.data);
		})();
	}, []);

	const uriSubmarino = 'https://www.submarino.com.br';

	function renderPage(image, index) {
		return (
			<View key={index}>
				<TouchableOpacity
					onPress={() =>
						Linking.openURL(uriSubmarino).catch((err) =>
							Alert.alert('Impossivel abrir a url', uriSubmarino)
						)}
				>
					<Image style={styles.image} source={{ uri: image.urlImagem }} />
				</TouchableOpacity>
			</View>
		);
	}

	return (
		<View style={styles.container}>
			{banners.length > 0 ? (
				<Carousel autoplay={false} index={0} pageSize={BannerWidth}>
					{banners.map((image, index) => renderPage(image, index))}
				</Carousel>
			) : (
				<Loading />
			)}
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		width: BannerWidth,
		height: BannerHeight,
		backgroundColor: '#ccc',
		justifyContent: 'center',
		elevation: 2
	},
	image: {
		width: BannerWidth,
		height: BannerHeight
	}
});
