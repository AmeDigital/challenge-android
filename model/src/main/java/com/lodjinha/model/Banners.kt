package com.lodjinha.model

data class Banners(val data: List<Banner>)
