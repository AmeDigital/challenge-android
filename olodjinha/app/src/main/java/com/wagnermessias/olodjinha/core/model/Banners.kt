package com.wagnermessias.olodjinha.core.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Banners (
    @SerializedName("data")
    val list: List<Banner>
):Serializable