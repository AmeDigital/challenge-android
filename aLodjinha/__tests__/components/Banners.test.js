import React from 'react';

import renderer from 'react-test-renderer';

import Banners from '../../src/components/Banners';

describe('Component Banners', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<Banners />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
