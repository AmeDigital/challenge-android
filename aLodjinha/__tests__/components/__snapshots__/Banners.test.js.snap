// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Banners should render without erros 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ccc",
      "elevation": 2,
      "height": 150,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#5e2a84"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
